### DISCUSSION TITLE
Avro schemas & helm

### DISCUSSION THREAD

**Discussion by swimand** [provectus/kafka-ui#1030] [Q&A] [answered]
Hi, 
Great work on the UI! I am currently trying it out as a replacement for Confluents alternative, and I am only really missing one feature, which you might have support for already. 

I have topics containing the schemas for the messages, automatically generated by with Avro. The schemas seem to be showing under the Schema view in the Kafka-UI, but I can't seem to get them to relate to the actual topics. 
In other words viewing the messages is currently not showing the correct format.

I tried adding the environment variable: 
`KAFKA_CLUSTERS_0_SCHEMANAMETEMPLATE: "%s-value"`
but it had no effect. I notived however that I have two schema topics per message type, one with the suffix `-value` and the second being `-key`.

Is it possible to connect the schemas with the topics?

**Comment by swimand** [comment, marked as answer]
I just found the solution myself. It seems that it is expected to also include the protocol when assigning the address to the schema registry. Since I have the schema registry running the same way I am running the kafka and zookeeper services, I simply assumed that one should exempt the protocol. My helm configuration ended up looking as follows, for those who are interested:

``` yaml

kafka-ui: 
  envs:
    config:
      KAFKA_CLUSTERS_0_NAME: "local"
      KAFKA_CLUSTERS_0_ZOOKEEPER: "local-zookeeper-svc:6990"
      KAFKA_CLUSTERS_0_BOOTSTRAPSERVERS: "local-kafka-svc:6991"
      KAFKA_CLUSTERS_0_SCHEMAREGISTRY: "http://local-schema-registry-svc:6992"
```
**Please note** that the name and ports on the services are not default confluent values!

**Comment by Haarolean** [comment]
Hi, thanks for this self Q&A! :D 

Also, it works like this: if you have a schema called "test" and a topic with the name "test-%something%" then the topic will use the corresponding schema.
